File: Day_17/Q67.java
import java.util.Scanner;

class Q67{

    static Scanner sc = new Scanner(System.in);

    static void input_array(int arr[],int n){
        for(int i = 0;i < n;i++){
            arr[i] = sc.nextInt();
        }
    }

    static void display_array(int arr[]){
        for(int i = 0;i < arr.length;i++){
            System.out.print(arr[i] + "\t");
        }
        System.out.println();
    }

    //Check Element Present
    static boolean is_present(int arr[],int size,int target){

        for(int i = 0;i < size;i++){
            if(arr[i] == target){
                return true;
            }
        }

        return false;
    }

    //Intersection Of Arrays
    static int[] intersection_array(int arr1[],int arr2[]){

        int temp[] = new int[Math.min(arr1.length,arr2.length)];

        int index = 0;

        for(int i = 0;i < arr1.length;i++){

            boolean found = false;

            //Check In Second Array
            for(int j = 0;j < arr2.length;j++){

                if(arr1[i] == arr2[j]){
                    found = true;
                    break;
                }
            }

            //Avoid Duplicate Elements
            if(found == true && is_present(temp,index,arr1[i]) == false){
                temp[index++] = arr1[i];
            }
        }

        int inter[] = new int[index];

        for(int i = 0;i < index;i++){
            inter[i] = temp[i];
        }

        return inter;
    }

    public static void main(String args[]){

        System.out.print("Enter Size Of First Array : ");
        int n1 = sc.nextInt();

        int arr1[] = new int[n1];

        System.out.print("Enter Elements Of First Array : ");
        input_array(arr1,n1);

        System.out.print("Enter Size Of Second Array : ");
        int n2 = sc.nextInt();

        int arr2[] = new int[n2];

        System.out.print("Enter Elements Of Second Array : ");
        input_array(arr2,n2);

        int inter[] = intersection_array(arr1,arr2);

        System.out.print("Intersection Array : ");
        display_array(inter);
    }
    sc.close();
}

File: Day_17/Q67.cpp
#include <bits/stdc++.h>
using namespace std;

void input_array(vector<int> &arr,int n){
    for(int i = 0;i < n;i++) cin >> arr[i];
}

void display_array(vector<int> &arr){
    for(int i = 0;i < arr.size();i++){
        cout << arr[i] << "\t";
    }
    cout << endl;
}

//Check Element Present
bool is_present(vector<int> arr,int target){

    for(int i = 0;i < arr.size();i++){
        if(arr[i] == target){
            return true;
        }
    }

    return false;
}

//Intersection Of Arrays
vector<int> intersection_array(vector<int> arr1,vector<int> arr2){

    vector<int> inter;

    for(int i = 0;i < arr1.size();i++){

        bool found = false;

        //Check In Second Array
        for(int j = 0;j < arr2.size();j++){

            if(arr1[i] == arr2[j]){
                found = true;
                break;
            }
        }

        //Avoid Duplicate Elements
        if(found == true && is_present(inter,arr1[i]) == false){
            inter.push_back(arr1[i]);
        }
    }

    return inter;
}

int main(){

    int n1,n2;

    cout << "Enter Size Of First Array : ";
    cin >> n1;

    vector<int> arr1(n1);

    cout << "Enter Elements Of First Array : ";
    input_array(arr1,n1);

    cout << "Enter Size Of Second Array : ";
    cin >> n2;

    vector<int> arr2(n2);

    cout << "Enter Elements Of Second Array : ";
    input_array(arr2,n2);

    vector<int> inter = intersection_array(arr1,arr2);

    cout << "Intersection Array : ";
    display_array(inter);

    return 0;
}
